history.go

```diff
@@ -46,24 +69,24 @@ func (h *History) LookUpInEpoch(key string, ep int64) (MerkleNode, *Authenticati
 	return n, ap, nil
 }
 
-func (h *History) GetSTR(ep int64) *SignedTreeRoot {
-	pointer := h.str
-	for pointer.epoch > ep && pointer != nil {
-		if pointer.prev == nil {
+func (h *History) GetSTR(ep Epoch) *SignedTreeRoot {
+	pointer := h.hashChain[h.currentSTR.epoch]
+	for pointer.epoch > ep && h.hashChain[pointer.epoch] != nil {
```

Comment:
> Since we have the epochs array, wouldn't it be easier to loop through that to find the right STR pointer?

We could do something like this:

```
func (h *History) GetSTR(ep Epoch) *SignedTreeRoot {
	if ep < h.epochs[0] {
		return nil
	}
	for i := 0; i < len(h.epochs)-1; i++ {
		if h.epochs[i] <= ep && ep < h.epochs[i+1] {
			return h.hashChain[h.epochs[i]]
		}
	}
	return h.hashChain[h.currentSTR.epoch]
}
```
but I think the current implementation looks more "natural" and is easier to understand. What do you think?